core/src/main/java/com/graphhopper/routing/util/parsers/FootPriorityParser.java

```diff
@@ -129,4 +129,12 @@ void collect(ReaderWay way, TreeMap<Double, PriorityCode> weightToPrioMap) {
         if (way.hasTag("bicycle", "official") || way.hasTag("bicycle", "designated"))
             weightToPrioMap.put(44d, SLIGHT_AVOID);
     }
+
+    protected boolean use_sidepath(ReaderWay way) {
+        return way.hasTag("foot", "use_sidepath")
+                || way.hasTag("sidewalk", "separate")
+                || way.hasTag("sidewalk:both", "separate")
+                || way.hasTag("sidewalk:left", "separate") && way.hasTag("sidewalk:right", sidewalksNoValues)
+                || way.hasTag("sidewalk:right", "separate") && way.hasTag("sidewalk:left", sidewalksNoValues);
```

Comment:
```suggestion
                || (way.hasTag("sidewalk:left", "separate") && way.hasTag("sidewalk:right", sidewalksNoValues))
                || (way.hasTag("sidewalk:right", "separate") && way.hasTag("sidewalk:left", sidewalksNoValues));
```

Reply:
The `&&` has precedence over `||`, but I'd still add the parenthesis because it makes it easier to spot.